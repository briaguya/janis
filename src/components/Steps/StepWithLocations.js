import React from 'react';
import PropTypes from 'prop-types';
import HtmlFromRichText from 'components/HtmlFromRichText';
import { useIntl } from 'react-intl';

const LocationInStep = ({ location }) => {
  const intl = useIntl();

  return (
    <a
      href={`/${intl.locale}/location/${location.locationPage.slug}/`}
      className="coa-StepWithLocations__location"
    >
      <div className="coa-StepWithLocations__location-titleaddress">
        <p className="coa-StepWithLocations__location-title">
          {location.locationPage.title}
        </p>
        <p className="coa-StepWithLocations__location-address">{`${
          location.locationPage.physicalStreet
        }${location.locationPage.physicalUnit}, ${
          location.locationPage.physicalCity
        }, ${location.locationPage.physicalState} ${
          location.locationPage.physicalZip
        }`}</p>
      </div>
      <div className="coa-StepWithLocations__location-arrow">
        <i class="material-icons">arrow_forward</i>
      </div>
    </a>
  );
};

const StepWithLocationsContainer = ({ description, locations }) => {
  return (
    <div className="coa-StepWithLocations__container">
      <HtmlFromRichText content={description} />
      {locations.map(location => (
        <LocationInStep location={location} />
      ))}
    </div>
  );
};

const StepWithLocations = ({ description, locations, singleStep }) =>
  singleStep ? (
    <StepWithLocationsContainer
      description={description}
      locations={locations}
    />
  ) : (
    <li>
      <StepWithLocationsContainer
        description={description}
        locations={locations}
      />
    </li>
  );

export default StepWithLocations;
