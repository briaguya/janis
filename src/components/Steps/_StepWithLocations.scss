.coa-StepWithLocations__container {
  flex-grow: 1;
  flex-shrink: 1;
  flex-basis: 0;

  p {
    margin-top: 0;
  }
}

.coa-StepWithLocations__location {
  display: flex;
  border: 1px solid #c6cace;
  padding: 1.4rem;
  border-radius: 4px;
  justify-content: space-between;
  text-decoration: none;
  &:hover {
    background: $coa-color-primary;
    .coa-StepWithLocations__location-title {
      color: $coa-color-white;
    }
    .coa-StepWithLocations__location-address {
      color: $coa-color-white;
    }
    .coa-StepWithLocations__location-arrow {
      color: $coa-color-white;
    }
  }
  &:focus {
    background: $coa-color-darker-blue;
    .coa-StepWithLocations__location-title {
      color: $coa-color-white;
    }
    .coa-StepWithLocations__location-address {
      color: $coa-color-white;
    }
    .coa-StepWithLocations__location-arrow {
      color: $coa-color-white;
    }
  }

  &:not(:last-child) {
    margin-bottom: 8px; // magic spacing number not used anywhere else
  }
}

.coa-StepWithLocations__location-title {
  margin: 0;
  font-size: 1.6rem;
  line-height: 2rem;
  font-weight: 600;
  color: #2d2e2f;
  margin-bottom: 0.8rem;
}

.coa-StepWithLocations__location-address {
  margin: 0;
  font-size: 1.4rem;
  line-height: 1.8rem;
  color: #3d4551;
}

.coa-StepWithLocations__location-arrow {
  color: #0049c7;
}
