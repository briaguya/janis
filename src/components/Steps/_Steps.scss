.coa-Steps {
  margin-bottom: $coa-spacing-large;
  padding-bottom: $coa-spacing-large;
}

.coa-rtl {
  .coa-Steps__list > ul > li {
    &:before {
      margin-left: $coa-spacing-md-small;
      margin-right: 0;
    }
  }
  .coa-Steps__list > ul {
    ol,
    ul {
      padding-left: 0;
      padding-right: $coa-spacing-xlarge + 2;
    }
  }
}

.coa-Steps__list {
  & > ol {
    counter-reset: step-item;
    list-style-type: none;
    margin-top: 20px;
    margin-bottom: 20px;
    margin-left: 20px;
    @include mobile {
      margin-left: 0;
    }

    & > li {
      align-items: flex-start;
      counter-increment: step-item;
      display: flex;
      flex-wrap: wrap;
      margin-left: 0;
      margin-bottom: 0;
      margin-top: 2.4rem;
      padding-bottom: 2.4rem;
      border-bottom: 1px solid $coa-color-grey;

      &:before {
        background-color: $coa-color-secondary-dark;
        border-radius: 50%;
        flex: 0 0 ($coa-line-height-medium + 0.2);
        color: $coa-color-primary;
        content: counter(step-item);
        font-size: $coa-font-size-md-small;
        font-weight: $coa-font-bold;
        line-height: $coa-line-height-medium + 0.2;
        text-align: center;
        margin-right: 100%;
        margin-bottom: 0;

        @include media($coa-medium-screen) {
          margin-right: $coa-spacing-large;
          margin-bottom: 0;
        }

        @include mobile {
          margin-bottom: 1.6rem;
        }

        // @include media($coa-large-screen) {
        //   font-size: $coa-font-size-md-small;
        //   line-height: $coa-line-height-md-small;
        //   flex: 0 0 ($coa-line-height-md-small + 0.2);
        // }
      }

      //TODO: adjust styles below when top level steps are not formatted within paragraphs
      & > p {
        display: inline;
        flex: 1 1;
        font-size: $coa-font-size-small;
        line-height: $coa-line-height-small;
        margin: 0;

        //p tag within p tag, wagtail issue?
        & > p {
          &:first-child {
            margin-top: 0;
          }
          font-size: inherit;
        }

        ul,
        li {
          font-size: inherit;
          // margin-bottom: 0;
        }
        // @include media($coa-medium-screen) {
        //   font-size: $coa-font-size-medium;
        //   line-height: $coa-line-height-medium;
        // }

        // @include media($coa-large-screen) {
        //   font-size: $coa-font-size-small;
        //   line-height: $coa-line-height-small;
        // }
      }
    }

    ol,
    ul {
      // padding-bottom: $coa-spacing-xlarge;
      padding-left: $coa-spacing-xsmall;
      padding-top: $coa-spacing-xsmall;
      list-style-type: disc;

      //TODO: adjust styles below when nested lists are not formatted within paragraphs
      p {
        margin: 0;
      }
    }
  }

  //TODO: remove styles below when steps are properly formatted JSON
  br {
    display: none;
  }
}
