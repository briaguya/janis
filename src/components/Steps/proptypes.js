import PropTypes from 'prop-types';

export const stepDetailGroupPropTypes = {
  steps: PropTypes.arrayOf(PropTypes.shape(stepDetailPropTypes)).isRequired,
};

export const stepDetailPropTypes = {
  title: PropTypes.string.isRequired,
  link: PropTypes.shape({
    url: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  }),
  content: PropTypes.string.isRequired,
};
