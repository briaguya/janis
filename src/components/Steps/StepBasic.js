import React from 'react';
import PropTypes from 'prop-types';
import HtmlFromRichText from 'components/HtmlFromRichText';
import { stepBasicPropTypes } from './proptypes';

const StepBasic = ({ stepAsHtmlFromRichText, singleStep }) =>
  singleStep ? (
    <p>
      <HtmlFromRichText content={stepAsHtmlFromRichText} />
    </p>
  ) : (
    <li>
      <p>
        <HtmlFromRichText content={stepAsHtmlFromRichText} />
      </p>
    </li>
  );

StepBasic.propTypes = stepBasicPropTypes;

export default StepBasic;
