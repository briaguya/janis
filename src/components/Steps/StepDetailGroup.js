import React from 'react';
import PropTypes from 'prop-types';

import { stepDetailGroupPropTypes } from './proptypes';

const StepDetailGroup = ({ steps }) => (
  <div className="coa-StepDetailGroup">StepDetailGroup</div>
);

StepDetailGroup.propTypes = stepDetailGroupPropTypes;

export default StepDetailGroup;
