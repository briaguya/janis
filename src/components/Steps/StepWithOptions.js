import React from 'react';
import PropTypes from 'prop-types';
import HtmlFromRichText from 'components/HtmlFromRichText';
import { optionPropTypes, stepWithOptionsPropTypes } from './proptypes';

import {
  Accordion,
  AccordionItem,
  AccordionItemHeading,
  AccordionItemButton,
  AccordionItemPanel,
  AccordionItemState,
} from 'react-accessible-accordion';

const StepOption = ({ option_name, option_description }) => (
  <div className="coa-StepOption">
    <AccordionItem className={'coa-AccordionItem'}>
      <AccordionItemHeading className={'coa-AccordionItemHeading'}>
        <AccordionItemButton className={'coa-AccordionButton'}>
          <h3 className="coa-StepOption__name">{option_name}</h3>
          <AccordionItemState>
            {({ expanded }) =>
              expanded ? (
                <i className="material-icons">remove</i>
              ) : (
                <i className="material-icons">add</i>
              )
            }
          </AccordionItemState>
        </AccordionItemButton>
      </AccordionItemHeading>
      <AccordionItemPanel className={'coa-AccordionPanel'}>
        <span>
          <HtmlFromRichText content={option_description} />
        </span>
      </AccordionItemPanel>
    </AccordionItem>
  </div>
);

StepOption.propTypes = optionPropTypes;

const StepWithOptionsContent = props => (
  <div className="coa-StepOption__container">
    <HtmlFromRichText content={props.description} />
    <Accordion className={'coa-Accordion'} allowMultipleExpanded={false}>
      {props.options.map(({ ...rest }, index) => (
        <StepOption key={index} {...rest} />
      ))}
    </Accordion>
  </div>
);

const StepWithOptions = ({ description, options, singleStep }) =>
  singleStep ? (
    <StepWithOptionsContent description={description} options={options} />
  ) : (
    <li className="coa-StepOption__item">
      <StepWithOptionsContent description={description} options={options} />
    </li>
  );

StepWithOptions.propTypes = stepWithOptionsPropTypes;

export default StepWithOptions;
