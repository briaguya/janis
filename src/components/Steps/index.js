import React from 'react';
import PropTypes from 'prop-types';

import StepBasic from './StepBasic';
import StepWithOptions from './StepWithOptions';
import { stepsPropTypes } from './proptypes';

const Steps = ({ steps }) => (
  <div className="coa-Steps">
    <div className="coa-Steps__list">
      <ul>
        {steps.map(step => {
          if (step.type === 'step_with_options_accordian')
            return (
              <StepWithOptions
                description={step.value.options_description}
                options={step.value.options}
              />
            );

          if (step.type === 'basic_step')
            return <StepBasic stepAsHtmlFromAdmin={step.value} />;
        })}
      </ul>
    </div>
  </div>
);

Steps.propTypes = stepsPropTypes;

export default Steps;
