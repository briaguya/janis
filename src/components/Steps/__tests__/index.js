import React from 'react';
import renderer from 'react-test-renderer';
// BrowserRouter is necessary so the tests know where to find Link
import { BrowserRouter } from 'react-router-dom';

import createComponentWithIntl from 'js/helpers/createComponentWithIntl';
import Steps from 'components/Steps';

jest.mock('react-accessible-accordion')

import AccordionWrapper from '..'

const {
  Accordion,
  AccordionItem,
  AccordionItemHeading,
  AccordionItemPanel,
  AccordionItemButton,
} = jest.requireActual('react-accessible-accordion')

describe('Step', () => {
  test('step renders, step length > 1', () => {
    const steps = [
      {
        type: "step_with_options_accordian",
        value: {
          options_description: "Determine how you would like to receive your certificate based on how quickly you need it",
          options: [
            {
              option_name:"3-5 business days by phone",
              option_description: "<ol><li>Call VitalChek, our secure online agent at 1-800-457-7586.</li><li>Verify your identity by answering a series of questions. If VitalChek cannot verify your identity with their standard questions, you will be instructed to upload a copy of your current government-issued identification and proof of shipping address before your application can be processed. This will delay the processing and delivery time.</li><li>Use a credit or debit card (Visa, MasterCard, American Express, or Discover) to pay for your certificate(s). Processing and UPS shipping fees will be added.</li></ol>"
            },
            {
              option_name: "2-3 weeks by mail",
              option_description: "<p>Mail the following items to Office of Vital Records, PO Box 1088, Austin, TX 78767-1088:</p>\
              <ol><li>If your home address is different than the address on your ID, also include a proof of residence like a utility bill, voter registration card, etc...</li><li>A self-addressed envelope with a stamp. You may use an express mail envelope for faster return delivery. If this is not included, you will not receive the certificate.</li></ol><p>If you need the certificate quickly through the mail, use an Express form of delivery, such as FedEx, USPS, or UPS, and send your application to: Office of Vital Records, 7201 Levander Loop, Bldg. C, Austin, TX 78702.</p>"
            }]
          },
        id:"f601e59c-9e52-46f3-947e-399120fa604a"
      },
      {
        type: "basic_step",
        value: "<p>Make sure you have enough money for the certificate(s) you need.</p><ul><li>The cost for each birth certificate is $23.</li><li>The cost for one death certificate is $21.</li></ul>",
        id:"2a910f7a-2063-4cb8-9f06-2d9b96c9f5cd"
      }
    ];
    const component = renderer.create(
      <BrowserRouter>
      	<Steps
          steps={steps}
      	/>
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('step renders, step length equals 1', () => {
    const step = [
      {
        type: "basic_step",
        value: "<p>Make sure you have enough money for the certificate(s) you need.</p><ul><li>The cost for each birth certificate is $23.</li><li>The cost for one death certificate is $21.</li></ul>",
        id:"2a910f7a-2063-4cb8-9f06-2d9b96c9f5cd"
      }
    ];
    const component = renderer.create(
      <BrowserRouter>
        <Steps
          steps={step}
        />
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});


