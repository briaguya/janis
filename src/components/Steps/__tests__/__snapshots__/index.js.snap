// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step step renders, step length > 1 1`] = `
<div
  className="coa-Steps"
>
  <div
    className="coa-Steps__list"
  >
    <ol>
      <li
        className="coa-StepOption__item"
      >
        <div
          className="coa-StepOption__container"
        >
          Determine how you would like to receive your certificate based on how quickly you need it
        </div>
      </li>
      <li>
        <p>
          <p>
            Make sure you have enough money for the certificate(s) you need.
          </p>
          <ul>
            <li>
              The cost for each birth certificate is $23.
            </li>
            <li>
              The cost for one death certificate is $21.
            </li>
          </ul>
        </p>
      </li>
    </ol>
  </div>
</div>
`;

exports[`Step step renders, step length equals 1 1`] = `
<div
  className="coa-Steps"
>
  <p>
    <p>
      Make sure you have enough money for the certificate(s) you need.
    </p>
    <ul>
      <li>
        The cost for each birth certificate is $23.
      </li>
      <li>
        The cost for one death certificate is $21.
      </li>
    </ul>
  </p>
</div>
`;
